docs/arcaflow/workflows/flow-control.md

```diff
@@ -2,6 +2,258 @@
+```yaml title="workflow.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        step_1_enabled:
+          type:
+            type_id: bool
+        step_1_input:
+          type:
+            type_id: integer
+steps:
+  my_step:
+    plugin:
+      deployment_type: image
+      src: path/to/my_plugin:1
+    input:
+      param_1: !expr $.input.step_1_input
+    enabled: !expr $.input.step_1_enabled
+outputs:
+  workflow_success:
+    plugin_output: !ordisabled $.steps.my_step.outputs.success
+```
+
+#### Alternative methods
+
+For handling disabled steps, `!oneof` and `!ordisabled` are the recommended methods because they cause output failure when the step fails. However, if it is acceptable for the workflow to succeed when a step crashes, the optional tags could be the right feature for your workflow.
+
+The alterative methods are the optional tags. The `oneof` tags instructed the workflow to build a `oneof_string` type with OR dependencies, requiring one of the options to have an output for the oneof to resolve. Meanwhile, the `optional` tags can resolve without an output by utilizing optional fields.
+
+| Tag              | Description                                      |
+|------------------|--------------------------------------------------|
+| `!wait-optional` | The typical one that should be used. It will resolve either when the value becomes present, or when it is known that the value will not resolve. |
+| `!soft-optional` | Not recommended for most use cases. It will not wait for the value to be ready. If the value is present, it will set the field. If it isn't, it will exclude it. |
+
+##### How to use `!wait-optional`
+
+```yaml title="workflow.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        step_1_enabled:
+          type:
+            type_id: bool
+        step_1_input:
+          type:
+            type_id: integer
+steps:
+  my_step:
+    plugin:
+      deployment_type: image
+      src: path/to/my_plugin:1
+    input:
+      param_1: !expr $.input.step_1_input
+    enabled: !expr $.input.step_1_enabled
+outputs:
+  workflow_success:
+    plugin_output: !wait-optional $.steps.my_step.outputs.success
+```
+
+If `my_step` is enabled and has an output, it will be present in the `plugin_output` field of the `workflow_success` output. Otherwise, the `workflow_success` output will not include the `plugin_output` field. The output will wait for the plugin to finish, if enabled.
+
+##### How to use `!soft-optional`
+
+!!! warning
+    This feature is not recommended for most use cases.
+
+The `!soft-optional` tag creates the loosest type of dependency. The output will __not__ wait for the step to finish or fail. This feature is only intended for fields that you want to get the output for if they succeed, but you are okay with it not being present if it doesn't finish.
```

Comment:
A good case for keeping Markdown lines relatively short; this paragraph is a single file line, which means I can't tie a comment to a specific section of the paragraph. The Markdown render code doesn't care about the line length, but shorter lines are a lot more convenient to review!

Anyway, the final sentence here is awkward. Maybe "Use this feature when you want to get output if the step succeeds, but you're happy to ignore it if the step doesn't finish."??

Reply:
I reworded that sentence, as well as a bit of the following paragraph.